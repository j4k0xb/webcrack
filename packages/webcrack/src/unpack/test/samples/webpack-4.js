!(function (e) {
  var a = {};
  function n(t) {
    if (a[t]) return a[t].exports;
    var r = (a[t] = { i: t, l: !1, exports: {} });
    return e[t].call(r.exports, r, r.exports, n), (r.l = !0), r.exports;
  }
  (n.m = e),
    (n.c = a),
    (n.d = function (e, t, i) {
      n.o(e, t) || Object.defineProperty(e, t, { enumerable: !0, get: i });
    }),
    (n.r = function (e) {
      'undefined' != typeof Symbol &&
        Symbol.toStringTag &&
        Object.defineProperty(e, Symbol.toStringTag, { value: 'Module' }),
        Object.defineProperty(e, '__esModule', { value: !0 });
    }),
    (n.t = function (e, t) {
      if ((1 & t && (e = n(e)), 8 & t)) return e;
      if (4 & t && 'object' == typeof e && e && e.__esModule) return e;
      var i = Object.create(null);
      if (
        (n.r(i),
        Object.defineProperty(i, 'default', { enumerable: !0, value: e }),
        2 & t && 'string' != typeof e)
      )
        for (var a in e)
          n.d(
            i,
            a,
            function (t) {
              return e[t];
            }.bind(null, a)
          );
      return i;
    }),
    (n.n = function (e) {
      var t =
        e && e.__esModule
          ? function () {
              return e.default;
            }
          : function () {
              return e;
            };
      return n.d(t, 'a', t), t;
    }),
    (n.o = function (e, t) {
      return Object.prototype.hasOwnProperty.call(e, t);
    }),
    (n.p = ''),
    n((n.s = 2));
})([
  ,
  function (e, t, i) {
    const a = i(3);
  },
  function (e, t, i) {
    const a = i(1);
    const module = 1;
    e.exports.color = '#FBC02D';
    {
      const module = 2;
      console.log(module);
      console.log(e);
    }
    t.a = 3;
  },
  function (e, t, i) {
    e.exports = 4;
  },
]);
