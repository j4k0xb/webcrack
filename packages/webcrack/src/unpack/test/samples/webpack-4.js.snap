WebpackBundle {
  "entryId": "2",
  "modules": Map {
    "1" => WebpackModule {
      "ast": const a = require("./3.js");,
      "id": "1",
      "isEntry": false,
      "path": "./1.js",
    },
    "2" => WebpackModule {
      "ast": const a = require("./1.js");
const _module = 1;
module.exports.color = "#FBC02D";
{
  const _module2 = 2;
  console.log(_module2);
  console.log(module);
}
exports.a = 3;,
      "id": "2",
      "isEntry": true,
      "path": "./index.js",
    },
    "3" => WebpackModule {
      "ast": module.exports = 4;,
      "id": "3",
      "isEntry": false,
      "path": "./3.js",
    },
  },
  "type": "webpack",
}